build/root/.kazelcfg.json

```diff
@@ -3,7 +3,8 @@
 	"SkippedPaths": [
 		"^_.*",
 		"/_",
-		"^third_party/etcd.*"
+		"^third_party/etcd.*",
+		"k8s.io/sample-apiserver/pkg/apis/.*"
```

Comment:
I think you're going to need similar changes to `Makefile.generated_files` to filter out the `k8s.io/sample-apiserver package` from it too,

Reply:
in particular, something around https://github.com/kubernetes/kubernetes/blob/5114d4e0b040d2f00417e0fd3e11204afd30f63c/build/root/Makefile.generated_files#L104-L107.

Reply:
We need to filter out https://github.com/kubernetes/kubernetes/tree/f8de7cea406a8d01799c4b4d40b892f3b38fa534/staging/src/k8s.io/code-generator/_examples as well. openapi is generated for it as well.

Reply:
Yes, good point. 

Also, it seems like `make generated_files` doesn't always run `openapi-gen` after tag changes. Filed #65928 for that one.

Reply:
Another thing which is not clear to me - do we need to exclude those packages (`staging/src/k8s.io/code-generator/_examples` and `k8s.io/sample-apiserver/pkg/apis`) only for `openapi-gen` or for `ALL_K8S_TAG_FILES`? Even before this PR, they were included in `deepcopy-gen`, `defaulter-gen`, `conversion-gen`.

Only `openapi-gen` is special, because it's outputting in `pkg/generated/openapi/zz_generated.openapi.go`

Reply:
We generate deepcopy, defaulters and conversion also through `hack/update-codegen.sh`. The resulting files are the same, so the overlap with `make generated_files` does not matter.

Reply:
I would leave it as it is now, for the given reason that deepcopy/defaulters/conversions are no problem. Moreover, this behaviour is not a regression of this PR. We had the same before. @ixdy are you ok with that? Can you approve the build/ change?

Reply:
@sttts the problem with this PR is that now `pkg/generated/openapi/zz_generated.openapi.go` will have the `Flundler` and `TestType`:

```bash
$ grep "Flunder" pkg/generated/openapi/zz_generated.openapi.go
                "k8s.io/sample-apiserver/pkg/apis/wardle/v1alpha1.Flunder":                                          schema_pkg_apis_wardle_v1alpha1_Flunder(ref),
                "k8s.io/sample-apiserver/pkg/apis/wardle/v1alpha1.FlunderList":                                      schema_pkg_apis_wardle_v1alpha1_FlunderList(ref),
                "k8s.io/sample-apiserver/pkg/apis/wardle/v1alpha1.FlunderSpec":                                      schema_pkg_apis_wardle_v1alpha1_FlunderSpec(ref),
                "k8s.io/sample-apiserver/pkg/apis/wardle/v1alpha1.FlunderStatus":                                    schema_pkg_apis_wardle_v1alpha1_FlunderStatus(ref),
                "k8s.io/sample-apiserver/pkg/apis/wardle/v1beta1.Flunder":                                           schema_pkg_apis_wardle_v1beta1_Flunder(ref),
                "k8s.io/sample-apiserver/pkg/apis/wardle/v1beta1.FlunderList":                                       schema_pkg_apis_wardle_v1beta1_FlunderList(ref),
                "k8s.io/sample-apiserver/pkg/apis/wardle/v1beta1.FlunderSpec":                                       schema_pkg_apis_wardle_v1beta1_FlunderSpec(ref),
                "k8s.io/sample-apiserver/pkg/apis/wardle/v1beta1.FlunderStatus":                                     schema_pkg_apis_wardle_v1beta1_FlunderStatus(ref),
                                        "disallowedFlunders": {
                                                        Description: "DisallowedFlunders holds a list of Flunder.Names that are disallowed.",
func schema_pkg_apis_wardle_v1alpha1_Flunder(ref common.ReferenceCallback) common.OpenAPIDefinition {
                                                        Ref: ref("k8s.io/sample-apiserver/pkg/apis/wardle/v1alpha1.FlunderSpec"),
                                                        Ref: ref("k8s.io/sample-apiserver/pkg/apis/wardle/v1alpha1.FlunderStatus"),
                        "k8s.io/apimachinery/pkg/apis/meta/v1.ObjectMeta", "k8s.io/sample-apiserver/pkg/apis/wardle/v1alpha1.FlunderSpec", "k8s.io/sample-apiserver/pkg/apis/wardle/v1alpha1.FlunderStatus"}
```

```bash
$ grep "TestType" pkg/generated/openapi/zz_generated.openapi.go
                "k8s.io/code-generator/_examples/apiserver/apis/example/v1.TestType":                                schema_apiserver_apis_example_v1_TestType(ref),
                "k8s.io/code-generator/_examples/apiserver/apis/example/v1.TestTypeList":                            schema_apiserver_apis_example_v1_TestTypeList(ref),
                "k8s.io/code-generator/_examples/apiserver/apis/example/v1.TestTypeStatus":                          schema_apiserver_apis_example_v1_TestTypeStatus(ref),
                "k8s.io/code-generator/_examples/apiserver/apis/example2/v1.TestType":                               schema_apiserver_apis_example2_v1_TestType(ref),
                "k8s.io/code-generator/_examples/apiserver/apis/example2/v1.TestTypeList":                           schema_apiserver_apis_example2_v1_TestTypeList(ref),
                "k8s.io/code-generator/_examples/apiserver/apis/example2/v1.TestTypeStatus":                         schema_apiserver_apis_example2_v1_TestTypeStatus(ref),
func schema_apiserver_apis_example_v1_TestType(ref common.ReferenceCallback) common.OpenAPIDefinition {
                                Description: "TestType is a top-level type. A client is created for it.",
                                                        Ref: ref("k8s.io/code-generator/_examples/apiserver/apis/example/v1.TestTypeStatus"),
                        "k8s.io/apimachinery/pkg/apis/meta/v1.ObjectMeta", "k8s.io/code-generator/_examples/apiserver/apis/example/v1.TestTypeStatus"},
func schema_apiserver_apis_example_v1_TestTypeList(ref common.ReferenceCallback) common.OpenAPIDefinition {
                                Description: "TestTypeList is a top-level list type. The client methods for lists are automatically created. You are not supposed to create a separated client for this one.",
                                                                                Ref: ref("k8s.io/code-generator/_examples/apiserver/apis/example/v1.TestType"),
                        "k8s.io/apimachinery/pkg/apis/meta/v1.ListMeta", "k8s.io/code-generator/_examples/apiserver/apis/example/v1.TestType"},
func schema_apiserver_apis_example_v1_TestTypeStatus(ref common.ReferenceCallback) common.OpenAPIDefinition {
func schema_apiserver_apis_example2_v1_TestType(ref common.ReferenceCallback) common.OpenAPIDefinition {
                                Description: "TestType is a top-level type. A client is created for it.",
                                                        Ref: ref("k8s.io/code-generator/_examples/apiserver/apis/example2/v1.TestTypeStatus"),
                        "k8s.io/apimachinery/pkg/apis/meta/v1.ObjectMeta", "k8s.io/code-generator/_examples/apiserver/apis/example2/v1.TestTypeStatus"},
func schema_apiserver_apis_example2_v1_TestTypeList(ref common.ReferenceCallback) common.OpenAPIDefinition {
                                Description: "TestTypeList is a top-level list type. The client methods for lists are automatically created. You are not supposed to create a separated client for this one.",
                                                                                Ref: ref("k8s.io/code-generator/_examples/apiserver/apis/example2/v1.TestType"),
                        "k8s.io/apimachinery/pkg/apis/meta/v1.ListMeta", "k8s.io/code-generator/_examples/apiserver/apis/example2/v1.TestType"},
func schema_apiserver_apis_example2_v1_TestTypeStatus(ref common.ReferenceCallback) common.OpenAPIDefinition {
```

I'll update the PR to filter-out those packages as @ixdy has suggested.

Reply:
> the problem with this PR is that now pkg/generated/openapi/zz_generated.openapi.go will have the Flundler and TestType:

ic. I thought that's fixed by your last change.